DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,325 @@
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaFail", "Failed replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
+    super(MoveReplica, self).__init__(operation, csPath)
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    self.rmsMonitoring = Operations().getValue("EnableActivityMonitoring", False)
+
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # own gMonitor stuff for files
+      gMonitor.registerActivity("ReplicateAndRegisterAtt", "Replicate and register attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateOK", "Replications successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateFail", "Replications failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterOK", "Registrations successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterFail", "Registrations failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaAtt", "Replica removals attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaOK", "Successful replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaFail", "Failed replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
 
     # Init ConsistencyInspector: used to check replicas
     self.ci = ConsistencyInspector()
 
-  def __call__( self ):
+  def __call__(self):
     """ call me maybe """
     # # check replicas first
     res = self.__checkReplicas()
     if not res["OK"]:
-      self.log.error( 'Failed to check replicas', res["Message"] )
+      self.log.error('Failed to check replicas', res["Message"])
 
     sourceSE = self.operation.SourceSE if self.operation.SourceSE else None
     if sourceSE:
       # # check source se for read
-      bannedSource = self.checkSEsRSS( sourceSE, 'ReadAccess' )
+      bannedSource = self.checkSEsRSS(sourceSE, 'ReadAccess')
       if not bannedSource["OK"]:
-        gMonitor.addMark( "ReplicateAndRegisterAtt", len( self.operation ) )
-        gMonitor.addMark( "ReplicateFail", len( self.operation ) )
+        if self.rmsMonitoring:
+          for opFile in self.operation:
+            for status in ["FileAttempted", "FileFailed"]:
```

Comment:
I do not get what you want to do here? You set all files FileAttempted and FileFailed This is what you want?

Reply:
Like I did this similar to `gMonitor` object as there the source is banned so we send status Attempted and Failed both as from here nothing will move further.

Reply:
@zmathe can I resolve this if we are clear on it?

Reply:
yes, you can resolve all of them